ee/control/client_http.go

```diff
@@ -40,77 +61,140 @@ func NewControlHTTPClient(logger log.Logger, addr string, client *http.Client, o
 	return c, nil
 }
 
-func (c *HTTPClient) Get(hash string) (data io.Reader, err error) {
-	verb, path := "GET", fmt.Sprintf("/api/v1/control/%s", hash)
+func (c *HTTPClient) GetConfig() (io.Reader, error) {
+	challengeReq, err := http.NewRequest(http.MethodGet, c.url("/api/agent/config").String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create challenge request: %w", err)
+	}
+
+	challenge, err := c.do(challengeReq)
+	if err != nil {
+		return nil, fmt.Errorf("could not make challenge request: %w", err)
+	}
+
+	configReq, err := http.NewRequest(http.MethodPost, c.url("/api/agent/config").String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not create config request: %w", err)
+	}
+	configReq.Header.Set(HeaderChallenge, string(challenge))
+	configReq.Header.Set("Content-Type", "application/json")
+	configReq.Header.Set("Accept", "application/json")
+
+	// Calculate first signature
+	localDbKeys := agent.LocalDbKeys()
+	if localDbKeys.Type() != "noop" {
```

Comment:
This would be better as an early return. And, honestly, as something throwing an error.

Reply:
Ah ok, my understanding was that maybe in the future it'd be possible for the request to be signed only by the hardware keys and not the db keys -- if that's the case, I could check both first to return an error if they're both noop? If not I'll just add an early return here.

Reply:
Updated to add an early return with an error here